iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -132,6 +133,16 @@ private async Task InitializeAndSetupClientAsync(CancellationToken cancellationT
                     cancellationToken: cancellationToken);
                 _logger.LogDebug($"The client instance has been opened.");
 
+                // The client retrieves twin values when it is initialized.
+                await RetryOperationHelper.RetryTransientExceptionsAsync(
+                    operationName: "GetTwin",
+                    asyncOperation: async () => await RetrieveTwinAsync(cancellationToken),
+                    shouldExecuteOperation: () => IsDeviceConnected,
+                    logger: _logger,
+                    exceptionsToBeIgnored: _exceptionsToBeIgnored,
+                    cancellationToken: cancellationToken);
+                _logger.LogDebug("The client has received twin values for the first time. ");

```

Comment:
nit: this block will be called any time the client is reinitialized and not just the first time, so let's update this log statement to something along the lines of:
_The client has retrieved the twin state after establishing connection to IoT hub_.